core/embed/trezorhal/stm32f4/backlight_pwm.c

```diff
@@ -16,6 +18,35 @@ static int pwm_period = 0;
 
 int backlight_pwm_set(int val) {
   if (BACKLIGHT != val && val >= 0 && val <= 255) {
+    // TPS61043: min 1% duty cycle
+    if (val < BACKLIGHT_PWM_TIM->ARR / 100) {
+      val = 0;
+    }
+
+    // TPS61043 goes to shutdown when duty cycle is 0 (after 32ms),
+    // so we need to set GPIO to high for at least 500us
+    // to wake it up.
+    if (BACKLIGHT_PWM_TIM->BACKLIGHT_PWM_TIM_CCR == 0) {
+      GPIO_InitTypeDef GPIO_InitStructure = {0};
+
+      HAL_GPIO_WritePin(BACKLIGHT_PWM_PORT, BACKLIGHT_PWM_PIN, GPIO_PIN_SET);
```

Comment:
Instead of reconfiguring GPIO, we **could** directly control OC1 by setting the `OC1M` bits in the `CCMR1` register. This applies to TIM1` used on the STM32F4 as well as to TIM8/17 used on the STM32U5.

Through the `OC1M` bits, OC1 can be forced high and then switched to PWM mode.

Reply:
there is a problem with this method actually

on TIM1 and TIM8 OC1M bits are preloaded, due to setting TIM_CR2_CCPC. 

TIM17 does not have this functionality, but we are still setting this CCPC bit (which is not breaking the rules of reference manual). The reaction of TIM17 on having this bit set is unfortunately that it doesn't update the OC1M value at all. We would need to allow setting this bit only for timers that support it, but we also already have boardloader and bootloader released, so we need to actually switch it off for some boards. 

For now, i don't want to waste more time on this so i will keep it as is, but will make an issue to fix this later.

Reply:
issue https://github.com/trezor/trezor-firmware/issues/3847